portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,33 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    private List<String> messages;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<>();
+    messages.add("Welcome to my protfolio");
+    messages.add("Have a great day :)");
+    messages.add("Hi there!");
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Tal!");
+    // Convert messages to JSON
+    String json = convertToJson(messages);
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
   }
+
+
+  private String convertToJson(List messages) {
```

Comment:
This method is redundant. Notice you can just do response.getWriter().println(new Gson().toJson(messages)), and have yourself a one-liner (if you prefer you can split the "new Gson()" to a param, but overall I'd keep it in doGet.